components/sup/src/templating/context.rs

```diff
@@ -95,81 +112,805 @@ impl<'a> RenderContext<'a> {
             "Census Group missing from list!",
         );
         RenderContext {
-            sys: sys,
-            pkg: pkg,
-            cfg: cfg,
+            sys: SystemInfo::from_sys(sys),
+            pkg: Package::from_pkg(pkg),
+            cfg: Cow::Borrowed(cfg),
             svc: Svc::new(census_group),
             bind: Binds::new(bindings, census),
         }
     }
+
+    // Exposed only for logging... can probably do this another way.
+    pub fn group_name(&self) -> &str {
+        self.svc.service_group.group()
+    }
 }
 
+////////////////////////////////////////////////////////////////////////
+// PRIVATE CODE BELOW
+////////////////////////////////////////////////////////////////////////
+
+// TODO (CM): Consider renaming this to something like SupInfo, since
+// it really captures information about how the Supervisor is invoked,
+// and consider exposing it under a "sup" key
+
+/// Templating proxy for a `manager::Sys` struct.
+///
+/// Currently exposed to users under the `sys` key.
 #[derive(Clone, Debug, Serialize)]
-pub struct Svc<'a> {
-    pub service: &'a str,
-    pub group: &'a str,
-    pub org: Option<&'a str>,
-    pub election_is_running: bool,
-    pub election_is_no_quorum: bool,
-    pub election_is_finished: bool,
-    pub update_election_is_running: bool,
-    pub update_election_is_no_quorum: bool,
-    pub update_election_is_finished: bool,
-    pub me: SvcMember<'a>,
-    pub first: SvcMember<'a>,
-    pub members: Vec<SvcMember<'a>>,
-    pub leader: Option<SvcMember<'a>>,
-    pub update_leader: Option<SvcMember<'a>>,
+struct SystemInfo<'a> {
+    version: Cow<'a, String>,
+    member_id: Cow<'a, String>,
+    ip: IpAddr,
```

Comment:
Why are the `IpAddr`s, ports and `permanent` not `Cow` types?

Reply:
I think it's because `IpAddr`, integers, and bool all implement `Copy` so this satisfied the compiler just fine. @cm I don't think memory really matters here since we're talking 1 to ~16 bytes per each of these types, but you do mean to point to something allocated elsewhere.

Reply:
Yeah, they all implement `Copy`... I'll make them `Cow`s for consistency (and future safety).